src/signal-handling.c

```diff
@@ -88,12 +93,134 @@ JL_DLLEXPORT void jl_exit_on_sigint(int on)
     exit_on_sigint = on;
 }
 
+void jl_show_sigill(void *_ctx);
+
 #if defined(_WIN32)
 #include "signals-win.c"
 #else
 #include "signals-unix.c"
 #endif
 
+void jl_show_sigill(void *_ctx)
+{
+#if defined(_CPU_X86_64_) || defined(_CPU_X86_)
+    uintptr_t pc = 0;
+#  if defined(_OS_LINUX_) && defined(_CPU_X86_64_)
+    pc = ((ucontext_t*)_ctx)->uc_mcontext.gregs[REG_RIP];
+#  elif defined(_OS_FREEBSD_) && defined(_CPU_X86_64_)
+    pc = ((ucontext_t*)_ctx)->uc_mcontext.mc_rip;
+#  elif defined(_OS_LINUX_) && defined(_CPU_X86_)
+    pc = ((ucontext_t*)_ctx)->uc_mcontext.gregs[REG_EIP];
+#  elif defined(_OS_FREEBSD_) && defined(_CPU_X86_)
+    pc = ((ucontext_t*)_ctx)->uc_mcontext.mc_eip;
+#  elif defined(_OS_DARWIN_)
+    pc = ((ucontext64_t*)_ctx)->uc_mcontext64->__ss.__rip;
+#  elif defined(_OS_WINDOWS_) && defined(_CPU_X86_)
+    pc = ((CONTEXT*)_ctx)->Eip;
+#  elif defined(_OS_WINDOWS_) && defined(_CPU_X86_64_)
+    pc = ((CONTEXT*)_ctx)->Rip;
+#  endif
+    // unsupported platform
+    if (!pc)
+        return;
+    uint8_t inst[15]; // max length of x86 instruction
+    int len = sizeof(inst);
+    // since we got a SIGILL and not a SIGSEGV or SIGBUS assume that
+    // the `pc` is pointed to valid memory.
+    // However, this does not mean that `pc + 14` is valid memory.
```

Comment:
Yes, I agree this could be factored into a few functions. Also `get_pc_from_ctx`.